cpp/src/replace/replace.cu

```diff
@@ -337,27 +309,88 @@ std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& inpu
 
 namespace { //anonymous
 
-template<typename Type, typename iter>
+template<typename Type, bool replacement_has_nulls>
 __global__
-void replace_nulls(cudf::size_type size,
-                   const Type* __restrict__ in_data,
-                   const cudf::bitmask_type* __restrict__ in_valid,
-                   iter replacement,
-                   Type* __restrict__ out_data)
-                   {
-  int tid = threadIdx.x;
-  int blkid = blockIdx.x;
-  int blksz = blockDim.x;
-  int gridsz = gridDim.x;
-
-  int start = tid + blkid * blksz;
-  int step = blksz * gridsz;
-
-  for (int i = start; i < size; i += step) {
-    out_data[i] = cudf::bit_is_set(in_valid, i) ? in_data[i] : replacement[i];
+void replace_nulls(cudf::column_device_view input,
+                   cudf::column_device_view replacement,
+                   cudf::mutable_column_device_view output,
+                   cudf::size_type* output_valid_count) {
+
+  cudf::size_type nrows = input.size();
+  cudf::size_type i = blockIdx.x * blockDim.x + threadIdx.x;
+
+  uint32_t active_mask = 0xffffffff;
+  active_mask = __ballot_sync(active_mask, i < nrows);
+  auto const lane_id { threadIdx.x % cudf::experimental::detail::warp_size };
+  uint32_t valid_sum { 0 };
+
+  while (i < nrows) {
+    uint32_t bitmask = 0xffffffff;
```

Comment:
This is only used inside the `replace_has_nulls` branch. Move the declaration there.

Reply:
Done. Didn't actually need to declare it at all, as it just got overwritten later.